<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Upload • animalcules</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Upload">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">animalcules</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.71</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Interactive Shiny Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/rmd/upload.html">1: Upload</a>
    </li>
    <li>
      <a href="../../articles/rmd/summary.html">2: Data Summary and Filtering</a>
    </li>
    <li>
      <a href="../../articles/rmd/abundance.html">3: Abundance</a>
    </li>
    <li>
      <a href="../../articles/rmd/diversity.html">4: Diversity</a>
    </li>
    <li>
      <a href="../../articles/rmd/dimension.html">5: Dimension Reduction</a>
    </li>
    <li>
      <a href="../../articles/rmd/differential.html">6: Differential Analysis</a>
    </li>
    <li>
      <a href="../../articles/rmd/biomarker.html">7: Biomarker</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../articles/animalcules.html">Command Line Analysis</a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/compbiomed/animalcules">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Upload</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/animalcules/blob/master/vignettes/rmd/upload.Rmd"><code>vignettes/rmd/upload.Rmd</code></a></small>
      <div class="hidden name"><code>upload.Rmd</code></div>

    </div>

    
    
<div id="example-data" class="section level3">
<h3 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example Data</h3>
<p>There are two example datasets available in animalcules: a synthetic toy dataset (already loaded) containing 50 samples and a real TB dataset containing 30 samples. Users could use any dataset to try all functions and features in animalcules.</p>
<p>Instructions:</p>
<ul>
<li>Use the pulldown menu to select the example dataset.</li>
<li>Click the “Upload” button.</li>
</ul>
<p>Running time: &lt; 0.5s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_030.png?raw=true" width="400"></p>
</div>
<div id="count-file" class="section level3">
<h3 class="hasAnchor">
<a href="#count-file" class="anchor"></a>Count File</h3>
<p>Both 16s and metagenomics generated data are supported here. The three required files are:</p>
<ul>
<li>Counts file: each row is a species/OTU, each column is a sample name.</li>
<li>Taxonomy file: each row is a species/OTU, each column is a taxonomy level.</li>
<li>Annotation file: each row is a sample name, each column is a variable/feature name.</li>
</ul>
<p>Make sure:</p>
<ul>
<li>The sample names in the counts file agree with the one in annotation file.</li>
<li>The microbe names in the counts file agree with the ones in the taxonomy file.</li>
<li>All three files have the same file separator / same file type (e.g. csv).</li>
</ul>
<p>Instructions:</p>
<ul>
<li>Click the “Browse…” button to upload required three files.</li>
<li>Input which column in the annotation file is the sample name (default is 1).</li>
<li>Check if the annotation file has header (default is TRUE).</li>
<li>Select the separator (default is comma AKA .csv file).</li>
<li>Click the button “Upload”.</li>
</ul>
<p>Note: Each uploaded table will show up in the right panel.</p>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_031.png?raw=true" width="300"></p>
</div>
<div id="pathoscope-file" class="section level3">
<h3 class="hasAnchor">
<a href="#pathoscope-file" class="anchor"></a>Pathoscope File</h3>
<p>To analyze pathoscope outputs, users need to upload pathoscope reports (use browser for multiple reports upload), as well as an annotation file containing metadata for each sample. Note that the sample name in the annotation file must match the non-suffix part of the pathoscope file name. For example, one pathoscope report filename is: “sample_011-sam-report.tsv”, then the corresponding sample name in the annotation file must be: “sample_011”.</p>
<p>Instructions:</p>
<ul>
<li>Click the “Browse…” button to upload required files.</li>
<li>Specify the pathoscope report file suffix (default is -sam-report.tsv).</li>
<li>Input which column in the annotation file is the sample name (default is 1).</li>
<li>Check if the annotation file has header (default is TRUE).</li>
<li>Select the separator (default is Tab AKA .tsv file).</li>
<li>Click the button “Upload”.</li>
</ul>
<p>Also, make sure to provide the correct column number for sample name in the annotation file, as well as the annotation file separator (tab, comma or semicolon).</p>
<p>Note: One example pathoscope eport table and the annotation table will show up in the right panel.</p>
<p>Running time:</p>
<ul>
<li>Test dataset A with 30 samples: 10.8s</li>
<li>Test dataset B with 587 samples: 22.1s</li>
</ul>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_032.png?raw=true" width="300"></p>
</div>
<div id="animalcules-id-file" class="section level3">
<h3 class="hasAnchor">
<a href="#animalcules-id-file" class="anchor"></a>animalcules-id file</h3>
<p>animalcules-id is a separate R pipeline that generates pathoscope-like outputs from fastq files. The required input is the animalcules-id generated .rds file. Here users could choose either EM count assay or Best hit assay.</p>
<p>Running time: &lt; 1s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_033.png?raw=true" width="400"></p>
</div>
<div id="animalcules-file" class="section level3">
<h3 class="hasAnchor">
<a href="#animalcules-file" class="anchor"></a>animalcules file</h3>
<p>In animalcules, users could choose to save their dataset to a .rds file in the Tab 2 (Summary and Filter). Later, users could load this saved dataset by uploading the .rds file to animalcules easily via this animalcules file upload option.</p>
<p>Running time: &lt; 0.5s</p>
<p><img src="https://github.com/compbiomed/materials/blob/master/animalcules/fig_034.png?raw=true" width="400"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Yue Zhao, Anthony Federico, W. Evan Johnson.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
